portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,33 +28,89 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private final List<Comment> sampleComments = new ArrayList<Comment>();
+  private final String UNSAFE_CHARACTERS_REGEX = "[^A-Za-z0-9._~()'!*:@,;+?\\s-]";
```

Comment:
I'm assuming this is to prevent users from putting in characters that don't match the encoding the website uses or something like that. A comment (or a more specific variable name) would be useful here so readers don't have to guess.

Reply:
Good point. This was my attempt to liberally block security concerns like code injections by just removing any character that is not one you'd normally expect to see in a comment. I'm going to take this as an opportunity to look more specifically into best practices for this and will add comments.

Reply:
So I did a bit of a deep dive into XSS and other potential security concerns regarding allowing user-submitted content to appear on webpages. Using innerText in the JS already prevents a consider amount of risk associated with XSS, but I decided to keep and clarify the whitelist as a way of blocking as much possibility as possible.